src/tagset.jl

```diff
@@ -11,7 +12,7 @@ struct TagSet
   data::TagSetStorage
   length::Int
   function TagSet() 
-    ts = TagSetStorage(ntuple(_ -> IntTag(0), Val(4)))
+    ts = TagSetStorage(ntuple(_ -> emptyIntTag, Val(maxTags)))
```

Comment:
So I'm betraying my ignorance of the Julia compiler here, but I just realized that Julia is not able to infer the type of `Val(maxTags)` properly. I believe the previous version would get inferred by constant propagation, but in theory the value of `maxTags` could get changed somewhere else and thus change the value here, so in principle the size wouldn't be inferrable. It would be better to define something like `const ValMaxTags = Val(4)` and use that instead.

I see `Val(maxTags)` is used elsewhere so probably I didn't consider this issue before, it should be change everywhere.

I just checked quickly switching to that design (and also using a corresponding `ValMaxTagLength = Val(8)` and I get the following speedup on `TagSet` construction. Current master branch:
```julia
julia> @btime TagSet($s)
  223.941 ns (10 allocations: 416 bytes)
"abcde,fghij,klmnop,qrstuv"
```
New design:
```julia
julia> @btime TagSet($s)
  103.391 ns (1 allocation: 80 bytes)
"abcde,fghij,klmnop,qrstuv"
```
which is pretty great speedup! I wrote the constructor pretty carefully and I remember wondering where all of those allocations were coming from.

I think there is over all a better design here, which would be defining a general parametrized type like `GenericTagSet{MaxNumTags, MaxTagLength}` and then hardcode `TagSet = TagSet{4, 8}`, and then extract `MaxNumTags` and `MaxTagLength` from the type within functions instead of from global variables. But we can start with the `const Val* = Val(*)` definitions for now.